contracts/DepositManager.sol

```diff
@@ -21,16 +23,24 @@ contract SubtreeQueue {
     uint256 public front = 1;
     uint256 public back = 0;
 
+    event DepositSubTreeReady(uint256 subtreeID, bytes32 subtreeRoot);
+
     function enqueue(bytes32 subtreeRoot) internal {
-        back += 1;
-        queue[back] = subtreeRoot;
+        uint256 subtreeID = back + 1;
+        back = subtreeID;
+        queue[subtreeID] = subtreeRoot;
+        emit DepositSubTreeReady(subtreeID, subtreeRoot);
```

Comment:
We forgot to add DepositSubTreeReady event in the `reenqueue` function. But if we add the event in the `enqueue` function, it comes with some benefits:
- we don't worry about forgetting it anywhere. 
- We don't pass the `subtreeID` and `readySubtree` all the way to the `depositFor`.